cnf-tests/testsuites/pkg/networks/sriov.go

```diff
@@ -0,0 +1,102 @@
+	sriovcluster "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/cluster"
+	. "github.com/onsi/gomega"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/nodes"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+
+	sriovtestclient "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/client"
+	sriovnamespaces "github.com/k8snetworkplumbingwg/sriov-network-operator/test/util/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/discovery"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/cnf-tests/testsuites/pkg/sriov"
+)
+
+func CleanSriov(sriovclient *sriovtestclient.ClientSet, namespace string) {
+	// This clean only the policy and networks with the prefix of test
+	err := sriovnamespaces.CleanPods(namespace, sriovclient)
+	Expect(err).ToNot(HaveOccurred())
+	err = sriovnamespaces.CleanNetworks(namespaces.SRIOVOperator, sriovclient)
+	Expect(err).ToNot(HaveOccurred())
+
+	if !discovery.Enabled() {
+		err = sriovnamespaces.CleanPolicies(namespaces.SRIOVOperator, sriovclient)
+		Expect(err).ToNot(HaveOccurred())
+	}
+	sriov.WaitStable(sriovclient)
+}
+
+func CreateSriovPolicyAndNetwork(sriovclient *sriovtestclient.ClientSet, namespace, networkName, resourceName, metaPluginsConfig string) {
+	sriovInfos, err := sriovcluster.DiscoverSriov(sriovclient, namespaces.SRIOVOperator)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(sriovInfos).ToNot(BeNil())
+
+	nn, err := nodes.MatchingOptionalSelectorByName(sriovInfos.Nodes)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(len(nn)).To(BeNumerically(">", 0))
+
+	sriovDevice, err := sriovInfos.FindOneSriovDevice(nn[0])
+	Expect(err).ToNot(HaveOccurred())
+
+	CreateSriovPolicy(sriovclient, sriovDevice, nn[0], resourceName)
+	CreateSriovNetwork(sriovclient, sriovDevice, namespace, networkName, resourceName, metaPluginsConfig)
+}
+
+func CreateSriovPolicy(sriovclient *sriovtestclient.ClientSet, sriovDevice *sriovv1.InterfaceExt, testNode, resourceName string) {
+	regularPolicy := &sriovv1.SriovNetworkNodePolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-policy",
+			Namespace:    namespaces.SRIOVOperator,
+		},
+
+		Spec: sriovv1.SriovNetworkNodePolicySpec{
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": testNode,
+			},
+			NumVfs:       4,
+			ResourceName: resourceName,
+			Priority:     99,
+			NicSelector: sriovv1.SriovNetworkNicSelector{
+				PfNames: []string{sriovDevice.Name},
+			},
+			DeviceType: "netdevice",
+		},
+	}
+
+	err := sriovclient.Create(context.Background(), regularPolicy)
+	Expect(err).ToNot(HaveOccurred())
+}
+
+func CreateSriovNetwork(sriovclient *sriovtestclient.ClientSet, sriovDevice *sriovv1.InterfaceExt, sriovNetworkNamespace, sriovNetworkName, resourceName, metaPluginsConfig string) {
```

Comment:
We have this already, see https://github.com/openshift-kni/cnf-features-deploy/blob/95301a57e61f7cd0fe835360d024cef1458bf96d/cnf-tests/testsuites/e2esuite/vrf/vrf_sriov.go#L85

Reply:
Moving to the network package for reuse.
The policy related code could also be shared, but a small update to the sriov operator repo is needed first.